### ISSUE TITLE
Feature request: multiple task servers, with conditional routing/syncing

### ISSUE THREAD

**Issue by extemporalgenome** [GothenburgBitFactory/taskwarrior#2319]
Related to #2156 in terms of expanding sharing/syncing capabilities, it would be empowering to have the ability to keep both personal and work tasks on the same device, managed with the same configuration, yet have work-related tasks sync to one task-server, while syncing personal tasks (or all tasks) to a separate task-server, based on some kind of task-based filter.

Conversely, I would want the ability to prevent certain devices (work laptop) from being able to receive personal tasks during a sync. Along with #2156, this could allow tasks to be shared with multiple colleagues, while keeping non-work tasks isolated, all based on task attributes (e.g. a "work" tag).

**Comment by 4kbt** [comment]
Agreed. This is the most-important feature gap that is preventing me from thriving with TaskWarrior.

In my naive perception of the TaskWarrior architecture, the essential ability is to be able to handle multiple backlog.data files (brought into TW with a virtual tag).   With that in place, there are myriad options for keeping the files from multiple sources in sync, whether through taskserver implementations or simply through a shared NFS/dropbox mount.

I am a longtime TW user. I have time over the next few weeks to attempt to dig into such a feature, if there are experienced devs to point me in the right direction.
